src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -519,6 +520,8 @@ private void copyDirectory(UnixPath source,
         try {
             mkdir(target, attrs.mode());
         } catch (UnixException x) {
+            if (x.errno() == EEXIST)
+                throw new FileSystemException(target.toString());
```

Comment:
Shouldn't this be `if (x.errno() == EEXIST && !flags.replaceExisting)` as FileAlreadyExistsException is okay when REPLACE_EXISTING not specified.

Reply:
I suppose the target could be created on another thread between when it is deleted and the call to `open` so this version would be better.

Reply:
So changed in bc43e5eb4579e8e5105d5af9d0a94ccdbb6b8eb1.